sw/device/tests/sim_dv/spi_tx_rx_test.c

```diff
@@ -231,95 +233,100 @@ static bool exp_irqs_fired(void) {
-        expected_irqs[kDifSpiDeviceIrqGenericRxWatermark]) {
-      expected_irqs[kDifSpiDeviceIrqGenericRxWatermark] = false;
-      LOG_INFO("SPI_DEVICE rx_above_level interrupt fired.");
-    }
-
-    // when 128 bytes received in RX_FIFO from SPI_HOST,
-    // read out and compare against the expected data
-    if (fired_irqs[kDifSpiDeviceIrqGenericRxWatermark] && !read_rx_fifo_done) {
-      size_t bytes_recved = 0;
-      uint8_t spi_device_rx_data[SPI_DEVICE_DATASET_SIZE];
-      CHECK_DIF_OK(dif_spi_device_recv(spi_device, spi_device_rx_data,
-                                       SPI_DEVICE_DATASET_SIZE, &bytes_recved));
-      if (bytes_recved == SPI_DEVICE_DATASET_SIZE) {
-        LOG_INFO("Received %d bytes from SPI_DEVICE RX_FIFO.", bytes_recved);
-        read_rx_fifo_done = true;
-      } else {
-        LOG_ERROR(
-            "SPI_DEVICE RX_FIFO recvd bytes mismatched: {act: %d, exp: %d}",
-            bytes_recved, SPI_DEVICE_DATASET_SIZE);
-      }
-      LOG_INFO("SPI_DEVICE read out RX FIFO.");
-      // expect SPI_HOST to send another 1024 bytes to fill RX SRAM FIFO
-      expected_irqs[kDifSpiDeviceIrqGenericTxUnderflow] = true;
-      fired_irqs[kDifSpiDeviceIrqGenericRxWatermark] = false;
-
-      // Check data consistency.
-      LOG_INFO("Checking the received SPI_HOST RX_FIFO data for consistency.");
-      for (int i = 0; i < SPI_DEVICE_DATASET_SIZE; ++i) {
-        CHECK(spi_device_rx_data[i] == exp_spi_device_rx_data[i],
-              "SPI_DEVICE RX_FIFO data[%d] mismatched: {act: %08x, exp: %08x}",
-              i, spi_device_rx_data[i], exp_spi_device_rx_data[i]);
-      }
-    }
-
-    if (read_rx_fifo_done && fired_irqs[kDifSpiDeviceIrqGenericTxUnderflow]) {
-      expected_irqs[kDifSpiDeviceIrqGenericRxWatermark] = true;
-      expected_irqs[kDifSpiDeviceIrqGenericTxUnderflow] = false;
-      LOG_INFO("SPI_DEVICE Tx underflow fired.");
-    }
-
-    if (read_rx_fifo_done && fired_irqs[kDifSpiDeviceIrqGenericRxWatermark]) {
-      expected_irqs[kDifSpiDeviceIrqGenericRxWatermark] = false;
-      expected_irqs[kDifSpiDeviceIrqGenericRxFull] = true;
-      LOG_INFO("SPI_DEVICE RX Above level interrupt fired.");
-    }
-
-    // After RX SRAM FIFO full, expect RX async FIFO overflow irq
-    if (fired_irqs[kDifSpiDeviceIrqGenericRxFull] &&
-        !fired_irqs[kDifSpiDeviceIrqGenericRxOverflow]) {
-      expected_irqs[kDifSpiDeviceIrqGenericRxFull] = false;
-      expected_irqs[kDifSpiDeviceIrqGenericRxOverflow] = true;
-      LOG_INFO("SPI_DEVICE RX_FIFO full interrupt fired.");
-    }
-
-    wait_for_interrupt();
-  }
+  // Set rx tx level back to default value so TxBelowLevel irq won't trigger.
+  IBEX_SPIN_FOR(fired_irqs[kDifSpiDeviceIrqGenericTxWatermark] &&
+                    expected_irqs[kDifSpiDeviceIrqGenericTxWatermark],
+                kTimeoutUs);
+
+  CHECK_DIF_OK(
+      dif_spi_device_set_irq_levels(spi_device, SPI_DEVICE_DATASET_SIZE, 0));
```

Comment:
I would put the comment in front of the `dif_spi_device_set_irq_levels` call
```suggestion
  IBEX_SPIN_FOR(fired_irqs[kDifSpiDeviceIrqGenericTxWatermark] &&
                    expected_irqs[kDifSpiDeviceIrqGenericTxWatermark],
                kTimeoutUs);

  // Set rx tx level back to default value so TxBelowLevel irq won't trigger.
  CHECK_DIF_OK(
      dif_spi_device_set_irq_levels(spi_device, SPI_DEVICE_DATASET_SIZE, 0));
```

Reply:
Thanks for the suggestion. Done